### PR TITLE
Extended operator errors and added forecast specific exceptions

### PR DESCRIPTION
**Pull request by prasankh**
- Introduced OperatorSchemaYamlError, ForecastSchemaYamlError and ForecastInputDataError.
- Made target_column and target_column mandatory.
- Updated code with relevant errors.

### CODE REVIEW

File: ads/opctl/operator/lowcode/forecast/schema.yaml [hunk 1]
```diff
@@ -196,7 +196,6 @@ spec:
     target_column:
       type: string
       required: true
-      default: target
```
Comment (ahosler) [inline ads/opctl/operator/lowcode/forecast/schema.yaml:199]: Why did we remove this? I think it's used in the ads opctl init method

File: ads/opctl/operator/lowcode/forecast/schema.yaml [hunk 2]
```diff
@@ -219,7 +218,7 @@ spec:
 
     target_category_columns:
       type: list
-      required: false
+      required: true
```
Comment (ahosler) [inline ads/opctl/operator/lowcode/forecast/schema.yaml:221]: Does the code break without specifying this series column? If so we should open a ticket to change that

**Review by ahosler — APPROVED** [review]
Looks awesome, thank you!
A couple of questions about the schema we can resolve later

File: ads/opctl/operator/operator_config.py [hunk 3]
```diff
@@ -13,7 +13,7 @@
 from ads.common.serializer import DataClassSerializable
 
 from .common.utils import OperatorValidator
-
+from .common.errors import OperatorSchemaYamlError
```
Comment (mrDzurb) [inline ads/opctl/operator/operator_config.py:16]: Let's replace this with absolute imports. 
```
from ads.opctl.operator.common.utils import OperatorValidator
from ads.opctl.operator.common.errors import OperatorSchemaYamlError
```

File: ads/opctl/operator/lowcode/forecast/schema.yaml (hunk 1 above)
Comment (prasankh) [reply ads/opctl/operator/lowcode/forecast/schema.yaml]: Okay, reverted this default value.

File: ads/opctl/operator/runtime/runtime.py [hunk 4]
```diff
@@ -15,7 +15,7 @@
 from ads.common.extended_enum import ExtendedEnum
 from ads.common.serializer import DataClassSerializable
 from ads.opctl.operator.common.utils import _load_yaml_from_uri
-
+from ..common.errors import OperatorSchemaYamlError
```
[the remark this answers is not in the document]
Comment (prasankh) [reply ads/opctl/operator/runtime/runtime.py]: Done

File: ads/opctl/operator/operator_config.py (hunk 3 above)
Comment (prasankh) [reply ads/opctl/operator/operator_config.py]: Done

File: ads/opctl/operator/lowcode/forecast/schema.yaml (hunk 2 above)
Comment (prasankh) [reply ads/opctl/operator/lowcode/forecast/schema.yaml:222]: Yes, It breaks and the error message is not meaningful hence made this mandatory.

**Review by codeloop — APPROVED** [review]

File: ads/opctl/operator/lowcode/forecast/schema.yaml (hunk 2 above)
Comment (mrDzurb) [reply ads/opctl/operator/lowcode/forecast/schema.yaml:222]: @ahosler @prasankh let's open a ticket to fix this, as Allen suggested?

**Comment by mrDzurb** [comment]
Looks like there are some conflicts with the target branch. We will need to merge the ```feature/forecasting``` into this branch at first.

**Review by mrDzurb — APPROVED** [review]

**Pull request merged**